### ISSUE TITLE
Reading editors content with a Screenreader

### ISSUE THREAD

**Issue by WestphalDenn** [hedgedoc/hedgedoc#1156]
**Description**

I as a blind user can not read the contents of the editor. The textfield appears to be empty. The only way to read the text is to select a portion of it which then only presents the selected content to me. 

Let me add that I would love to be able to use HedgeDoc like everyone else is in our company. Thanks fo rhelping! 

**To Reproduce**

*Log in to HedgeDoc
* Open a document of your choice. 
* Using a Screenreader (NVDA for example) enter the Editor. 
* Using the arrow-keys try to read the text. 
* Enter new text or try to delete some. 

**Expected behavior**
At all times text that I pass should be seen by the Screenreader so I can work on notes. not only alone but also colaborative with other people. But that only works if I get Feedback on what changes in a document and what it consists of.

**Comment by DerMolly** [comment]
Hey @WestphalDenn,

I'm really sorry to hear this. The problem is unfortunately not really in our hand. The problem lies with CodeMirror (the editor we use see https://github.com/codemirror/CodeMirror/issues/4604). Apparently CodeMirror 6 should be much more accessible, but it's currently not finished and when it will be, we'll incorporate it into the upcoming 2.0 and won't back port this change to 1.x code. So it seems HedgeDoc won't be usable for user in need of ScreenReaders and the like until 2.0 comes out. Depending on when CodeMirror 6 releases and how many changes we need to make for it to work it could take even longer.
But be sure we're aware of that problem and want to fix this.

**Comment by DougInAMug** [comment]
From https://codemirror.net/6/

> The project is in the beta phase—there's a stable interface, but small breaking changes might still happen. It is being used in production in a few mid-size systems, but is not as battle-tested as the old version.

**Comment by WestphalDenn** [comment]
Thank you for your reply. The Link you have provided me is indeed accessible when using a screen reader. 

 

-- 

Dennis Westphal // Trainer + Consultant 

 ***@***.***> ***@***.*** // Tel: +49 30 28040-0 

 

GzEvD - Gesellschaft zur Entwicklung von Dingen mbH // Osloer Str. 17 // D-13359 Berlin 

Amtsgericht Charlottenburg HRB 140685 B // Geschäftsführer: Stephan Luckow 

GDPR "mailbox":  <https://kurz.gzevd.de/briefkasten> https://kurz.gzevd.de/briefkasten

 

Von: Erik Michelson ***@***.***> 
Gesendet: Dienstag, 12. April 2022 10:28
An: hedgedoc/hedgedoc ***@***.***>
Cc: WestphalDenn ***@***.***>; Mention ***@***.***>
Betreff: Re: [hedgedoc/hedgedoc] Reading editors content with a Screenreader (#1156)

 

As stated above, this will be fixed with CodeMirror 6.
Due to the amount of work that needs to be done to switch from CodeMirror 5 to CodeMirror 6, we're not fixing this in HedgeDoc 1.x.

However, the UI for HedgeDoc 2 already uses CodeMirror 6 and therefore shouldn't have this problem anymore.
You can testdrive it here: https://hedgedoc.dev/n/features
